src/Corax/Utils/EntryIdEncodings.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+namespace Corax.Utils;
+
+public static class EntryIdEncodings
+{
+    //Schema:
+    //[CONTAINER_ID][FREQUENCY][CONTAINER_TYPE]
+    
+    // Positions:
+    private const int FrequencySizeInBits = 8;
+    private const int EntryIdOffset = FrequencySizeInBits + ContainerTypeOffset;
+    private const long Mask = 0xFFL;
+    private const int ContainerTypeOffset = 2;
+
+    
+    
+    private const int AllBuffer = -1;
+    
+    // Quantization parameters:
+    private const long Min = 0;
+    private const long Max = 255;
+    private const long QuantizationMax = 256 - 1;
+    // ReSharper disable once PossibleLossOfFraction
+    private const double QuantizationStep = (Max - Min) / (double)QuantizationMax;
```

Comment:
I used this for testing, but I believe we've to bump `QuantizationMax` :)

Reply:
What is the meaning of Quantization in this context?

Reply:
Since we've only 8 bits we want to encode much information as we can. Maybe it's worth encoding the range 0-1024 in 8 bits? That means we encode [0,1,2,3] as 0 etc. I would like to do some tests later about that. Please notice I tested ranking only on "small" indexes, so now encoding only in range <0;255> maybe is good but will not be for bigger indexes.

Reply:
Didn't @redknightlois discuss this here: https://www.cs.otago.ac.nz/research/student-publications/MCrane2013-2.pdf

8 bits should be sufficient.

For that matter, we can encode that using exponent.

So do something like `32 - lzcnt` on storage, which will give you small range.
And then `1 << storeFreq` to get the actual (rough) value?
That would give us frequencies in the power of two range and is really cheap to compute.

Reply:
Yes, we should use exponential quantilization, not linear. It is effective, easy to compute and we can get away with it with very small amount of bits.

Reply:
done